crates/wac-resolver/src/registry.rs

```diff
@@ -45,32 +48,34 @@ impl RegistryPackageResolver {
     /// client configuration file.
     ///
     /// If `url` is `None`, the default URL will be used.
-    pub fn new(url: Option<&str>, bar: Option<Box<dyn ProgressBar>>) -> Result<Self> {
+    pub async fn new(
+        url: Option<&str>,
+        bar: Option<Box<dyn ProgressBar>>,
+        retry: Option<Retry>,
```

Comment:
nit: document the `retry` arg.

Reply:
Sorry, by "comment" I meant a document so that consumers can see why the `retry` param is used for.